solutions/fully-configurable/variables.tf

```diff
@@ -0,0 +1,154 @@
+  default     = []
+}
+
+variable "cloud_monitoring_plan" {
+  type        = string
+  description = "The IBM Cloud Monitoring plan to provision. Available values are `lite` and `graduated-tier`."
+  default     = "graduated-tier"
+
+  validation {
+    condition     = can(regex("^lite$|^graduated-tier$", var.cloud_monitoring_plan))
+    error_message = "Specify one of the following values for the `cloud_monitoring_plan`: `lite` or `graduated-tier`."
+  }
+}
+
+variable "enable_platform_metrics" {
+  type        = bool
+  description = "When set to `true`, the IBM Cloud Monitoring instance collects the platform metrics."
+  default     = false
+}
+
+########################################################################################################################
+# Metrics Routing
+########################################################################################################################
+
+variable "metrics_routing_target_name" {
+  type        = string
+  description = "The name of the IBM Cloud Metrics Routing target where metrics are collected. If the prefix variable is passed, the name of the target is prefixed to the value in the `<prefix>-value` format."
+  default     = "cloud-monitoring-target"
+}
+
+variable "metrics_routing_route_name" {
+  type        = string
+  description = "The name of the IBM Cloud Metrics Routing route for the default route that indicate what metrics are routed in a region and where to store them. If the prefix variable is passed, the name of the target is prefixed to the value in the `<prefix>-value` format."
+  default     = "metrics-routing-route"
+}
+
+variable "enable_metrics_routing_to_cloud_monitoring" {
+  type        = bool
+  description = "Whether to enable metrics routing from IBM Cloud Metric Routing to Cloud Monitoring."
+  default     = true
+}
+
+variable "metrics_router_routes" {
+  type = list(object({
+    name = string
+    rules = list(object({
+      action = string
+      targets = list(object({
+        id = string
+      }))
+      inclusion_filters = list(object({
+        operand  = string
+        operator = string
+        values   = list(string)
+      }))
+    }))
+  }))
+  default     = []
+  description = "Routes for IBM Cloud Metrics Routing. [Learn more](https://github.com/terraform-ibm-modules/terraform-ibm-observability-da/blob/main/solutions/instances/DA-types.md#metrics-router-routes-)"
+}
+
+variable "cbr_rules" {
+  type = list(object({
+    description = string
+    account_id  = string
+    rule_contexts = list(object({
+      attributes = optional(list(object({
+        name  = string
+        value = string
+    }))) }))
+    enforcement_mode = string
+    operations = optional(list(object({
+      api_types = list(object({
+        api_type_id = string
+      }))
+    })))
+  }))
+  description = "(Optional, list) List of context-based restrictions rules to create"
```

Comment:
This needs to point to the markdown helper doc

Reply:
Going to merge the PR as is and we can add this in a subsequent PR. Tracked in https://github.com/terraform-ibm-modules/terraform-ibm-cloud-monitoring/issues/26